src/VirtualMonoRepo/README.template.md

```diff
@@ -139,11 +139,10 @@ ln -s $HOME/.dotnet/dotnet /usr/bin/dotnet
 
 You can also utilize [GitHub Codespaces](https://github.com/features/codespaces) where you can find preset containers in this repository.
 
-### Exporting a source archive
+### Building outside of git
 
-In case you'd like to export a more lightweight archive of sources that can be built outside of this git repository, a simple copy of the working tree won't do.
-The build is using some git metadata (information from the `.git` directory) that are needed to be kept with the sources.
-To export a `tar.gz` archive of the sources, you need to use the `eng/pack-sources.sh` script from within a clone of the VMR checked out at the revision that you're interested in.
+.NET uses git metadata so that it can link assemblies to their original source code when debugging (think "Step into.." functionality) and for that it needs information about the original place the code comes from.
+When you're building source code only, taken outside of context of a git repository (e.g. you download it from the release page), you will need to specify the source repository to the `prep.sh` script via the `--source-repository` and `--source-version` arguments. This can be your fork of the repository and should match the origin where your SDK was built from.
```

Comment:
What is the UX if someone builds without a .git directory and doesn't call prep?  Does this result in an error or no source-link info?  I think there are a number of distro maintainer build scenarios that don't involve prep.sh - they bring their own SDK and artifacts.  Initially I was expecting the parameters would be required for the build script and not part of prep.  That was based on two things.  1. I was hopping we could flow this information to source-link outside of git.  2. I was expecting we would want build to be where an error would occur if the metadata was not specified.

Reply:
Oh, I didn't know not calling `prep.sh` is a feasible flow too.

1. I originally wanted to flow it like you describe but it seems that sourcelink is not the only repo that needs this kind of information. For instance `source-build-reference-packages` hits a problem with missing `.git` even before sourcelink does. I don't know how many other places like that are down the line. I imagine there might be new appearing in future so it just seemed better to create the two files right away.

2. It makes sense and I have originally put it there but the fact that `build.sh` is used to run smoke tests too somehow gets in a way of that. What would the UX be for `build.sh`? You'd need to provide this for building only? Or should we rather move the smoke test functionality into `test.sh`? The way the script is written is already sort of this way - that these are two quite different exclusive actions.

Reply:
I think the dependency on `.git` needs more investigation.  Can you share the problem encountered with SBRP?  To be clear, it is the use of sourcelink while building each repo that I expect to depend on the `.git` directory.  This issue will not be specific to just the sourcelink repo.

Reply:
You're right, it's the sourcelink functionality in each of the repo we try to build. The problem comes from here: https://github.com/dotnet/dotnet/blob/d9b8caf4288bd0546d395498678a101b492c66eb/src/sourcelink/src/Microsoft.Build.Tasks.Git/build/Microsoft.Build.Tasks.Git.targets#L25

I spent some time looking into this and trying to disable this behaviour but it's a mix of 3 layers:
- Arcade
- Source Link
- SDK

It's both how the targets are intertwined and also a question how to pass the properties down through all those layers. I believe this will require a change in either Arcade or SourceLink or both, However, I don't know enough about these repos at the moment so will require more time.

Reply:
How are you thinking about fixing this intertwined issue?  Should a separate product issue be logged here to support this new scenario?  For now we check in these proposed changes.  Once fixed, we can refactor source-build to take advantage - e.g. stop creating the .git files, move the new options from prep to the build scrip, and flow the source-link options via the environment/build configuration.  Alternatively are you thinking of keeping this PR opened until the intertwined issue is resolved?

Reply:
Yes, I am in agreement we should merge this PR and log a sourcelink issue tracking the additional enhancements.

Reply:
https://github.com/dotnet/source-build/issues/3359